pkg/sdk/sdk_test.go

```diff
@@ -0,0 +1,518 @@
+}
+func Test_Register(t *testing.T) {
+	tests := []struct {
+		name              string
+		setupFunc         func() *httptest.Server
+		mockServerHandler func(w http.ResponseWriter, r *http.Request)
+		ValidateFunc      func(id string, err error)
+		cleanupFunc       func(*httptest.Server)
+		expectedResponse  model.Response
+	}{
+		{
+			name: "Success:: Register",
+			setupFunc: func() *httptest.Server {
+				svr := testServer("/v1/register", http.MethodPost, func(w http.ResponseWriter, r *http.Request) {
+					err := r.ParseMultipartForm(10000) //File size to come from config
+					if err != nil {
+						response.ToJson(w, http.StatusBadRequest, "file size exceeded", nil)
+						t.Error(err.Error())
+						return
+					}
+					file, _, err := r.FormFile("file")
+					if err != nil {
+						response.ToJson(w, http.StatusBadRequest, "file parse fail", nil)
+						t.Error(err.Error())
+						return
+					}
+					fileBytes, err := ioutil.ReadAll(file)
+					if err != nil {
+						response.ToJson(w, http.StatusBadRequest, "file read fail", nil)
+						t.Error(err.Error())
+						return
+					}
+					if !reflect.DeepEqual(fileBytes, []byte("abc")) {
+						t.Errorf("Want: %v, Got: %v", []byte("abc"), fileBytes)
+					}
+					defer file.Close()
+					cType := r.Header.Get("Content-Type")
+					x := strings.Split(cType, ";")
+					if !reflect.DeepEqual(x[0], "multipart/form-data") {
+						t.Errorf("Want: %v, Got: %v", "multipart/form-data", x[0])
+					}
+					response.ToJson(w, http.StatusCreated, "SUCCESS", map[string]interface{}{
+						"id": "1",
+					})
+				})
+				return svr
+			},
+			ValidateFunc: func(id string, err error) {
+				if err != nil {
+					t.Errorf("Want: %v, Got: %v", nil, err.Error())
+				}
+				if id != "1" {
+					t.Errorf("Want: %v, Got: %v", "not nil", "")
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+		{
+			name: "Failure :: Register :: id not found in response",
+			setupFunc: func() *httptest.Server {
+				svr := testServer("/v1/register", http.MethodPost, func(w http.ResponseWriter, r *http.Request) {
+					response.ToJson(w, http.StatusCreated, "SUCCESS", map[string]interface{}{
+						"xyz": "1",
+					})
+				})
+				return svr
+			},
+			ValidateFunc: func(id string, err error) {
+				if err.Error() != errors.New("id not found in response").Error() {
+					t.Errorf("Want: %v, Got: %v", "id not found in response", err.Error())
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+		{
+			name: "Failure:: Register:: failed to assert response data",
+			setupFunc: func() *httptest.Server {
+				svr := testServer("/v1/register", http.MethodPost, func(w http.ResponseWriter, r *http.Request) {
+					response.ToJson(w, http.StatusCreated, "SUCCESS", "1")
+				})
+				return svr
+			},
+			ValidateFunc: func(id string, err error) {
+				if err.Error() != errors.New("unable to parse response data").Error() {
```

Comment:
Why is a new error created when only the string(message) is being compared?

Reply:
done